### ISSUE TITLE
[instancer] Update for avar2

### ISSUE THREAD

**Issue by behdad** [fonttools/fonttools#3086]

**Comment by behdad** [comment]
This might be hard.

**Comment by anthrotype** [comment]
can you elaborate why it might be hard?

**Comment by justvanrossum** [comment]
Here is an idea (that may not be a _general_ solution): I have found that is possible to recreate the .designspace-style mappings from avar2 (at least as long avar2 was built from such mappings). From there it should be possible to slice/instance the mappings, similarly how outline data is sliced/instantiated, and then rebuild avar2.

**Comment by behdad** [comment]
> Here is an idea (that may not be a _general_ solution): I have found (*) that it is possible to recreate the .designspace-style mappings from avar2 (at least as long as avar2 was built from such mappings).

@Lorp and I were just talking about this. Here's how the reverse conversion can fail even if it was generated using varLib.models:

Here's the mappings:
- wght=0 -> wgt2=0
- wght=.5 -> wgt2=0
- wght=1. -> wgt2=1.

Here there are two tents, at wght=.5 and wght=1.; however, since the delta at wght=.5 is zero, it won't be encoded in the ItemVariationStore. So your code won't distinguish it from:
- wght=0 -> wgt2=0
- wght=1. -> wgt2=1.

We need to consider the min/max of the tents as well somehow. And then filter out the unnecessary masters? I'll think more about it.

**Comment by justvanrossum** [comment]
Ah, good catch. Looking at my code, it will be relatively easy to take the tent min/max into account. It indeed currently only looks at [PeakCoord](https://github.com/googlefonts/fontra/blob/574354d09feaf1b3d6c503c4ec79c5436a8f0429/src/fontra/backends/opentype.py#L182).

**Comment by behdad** [comment]
> it will be relatively easy to take the tent min/max into account.

Right. The question then is do we get too many extra mappings? But it should at least work.

**Comment by Lorp** [comment]
Will it work to infer 2^n+1 mappings from every region encoded in the IVS? (where n=number of dimensions with delta!=0, thus encoding all of the region’s corners, and the +1 is for the peak). Then we remove duplicates.

**Comment by Lorp** [comment]
Apologies for multiple edits of above. I think I’m done :)
